src/contexts/providers.context.tsx

```diff
@@ -49,35 +61,57 @@ const ProvidersProvider: FC = (props) => {
       setAccount({ status: "loading" });
       switch (walletName) {
         case WalletName.METAMASK: {
-          if (window.ethereum && window.ethereum.isMetaMask) {
-            const web3Provider = new Web3Provider(window.ethereum);
-            return getConnectedAccounts(web3Provider)
-              .then((accounts) => {
-                setConnectedProvider(web3Provider);
-                setAccount({ status: "successful", data: accounts[0] });
-              })
-              .catch((error) => {
-                if (isMetamaskUserRejectedRequestError(error)) {
-                  setAccount({ status: "pending" });
-                } else {
-                  void parseError(error).then((parsed) => {
-                    openSnackbar({
-                      type: "error",
-                      parsed,
-                    });
-                  });
-                }
-              });
-          } else {
+          if (env === undefined) {
             return setAccount({
               status: "failed",
-              error: `We cannot detect your wallet. Make sure the ${WalletName.METAMASK} extension is installed and active in your browser.`,
+              error: "The env has not been initialized correctly",
             });
           }
+
+          try {
+            if (window.ethereum && window.ethereum.isMetaMask) {
+              const web3Provider = new Web3Provider(window.ethereum);
+              const requestedNetwork = await web3Provider.getNetwork();
+              const supportedNetworks = await Promise.all(
+                env.chains.map((chain) => chain.provider.getNetwork())
+              );
+              const requestedChainId = requestedNetwork.chainId;
+              const supportedChainIds = supportedNetworks.map((network) => network.chainId);
+
+              if (!supportedChainIds.includes(requestedChainId)) {
+                return setAccount({
+                  status: "failed",
+                  error: "Switch your network to Ethereum or Polygon Hermez to continue",
+                });
+              }
+
+              const accounts = await getConnectedAccounts(web3Provider);
+
+              setConnectedProvider(web3Provider);
+              return setAccount({ status: "successful", data: accounts[0] });
+            } else {
+              return setAccount({
+                status: "failed",
+                error: `We cannot detect your wallet. Make sure the ${WalletName.METAMASK} extension is installed and active in your browser`,
+              });
+            }
+          } catch (error) {
+            const errorMessage = await parseError(error);
+
+            if (!isMetamaskUserRejectedRequestError(error)) {
```

Comment:
In which case are we going to get a `MetamaskUserRejectedRequestError` considering that we're not asking for any signature in the login process?

Reply:
If you have not logged in yet to the app, you need to choose which account you want to connect. In that scenario you can click on the "Cancel" button and get this error.

Reply:
Do you mean this "account connection" screen? I'm unable to reproduce the case...

![image](https://user-images.githubusercontent.com/15896177/166212812-27cd67d8-4c45-4f48-9d9d-334a002f5b6c.png)

Reply:
Got it. Here: 
![image](https://user-images.githubusercontent.com/15896177/166217822-8aef13d6-af82-4402-a6b0-677fd501c35a.png)